packages/analytics/src/integrations/Omnitracking/definitions.ts

```diff
@@ -600,20 +603,27 @@ export const trackEventsMapper: Readonly<OmnitrackingTrackEventsMapper> = {
     actionArea: data.properties?.from,
     productId: getProductId(data.properties),
     lineItems: getProductLineItems(data),
+    moduleTitle: JSON.stringify([data.properties?.list]),
```

Comment:
To avoid these duplications of moduleTitle and Id, would be more secure create an helper to deal with this module, then call the helper function in each event.

In this method you can also check if this property are undefined after JSON.stringify to avoid "[undefined]" values.

Also, check what would be the correct behaviour in undefined properties situation (if fill with empty array, or if assign the property with undefined instead of empty array)

Reply:
Fixed!
Created a helper function to avoid duplications, but also check if the parameters are undefined and check if it's a non recommendations product details event (in this scenario it's not necessary to send these parameters)!
Data team agreed that in case _moduleId_ or _moduleTitle_ are undefined, it should be sent an empty array